go/libkb/burst_cache.go

```diff
@@ -0,0 +1,96 @@
+package libkb
+
+import (
+	"fmt"
+	lru "github.com/hashicorp/golang-lru"
+	"golang.org/x/net/context"
+	"time"
+)
+
+// BurstCachce is an LRU+SingleFlighter useful for absorbing short-lived bursts
+// of lookups. If multiple goroutines are fetching resource A, they all will block on
+// the first fetch, and can (if the fetch returns without error) share the answer
+// of the first request.
+type BurstCache struct {
+	Contextified
+	locktab   LockTable
+	lru       *lru.Cache
+	cacheLife time.Duration
+	name      string
+}
+
+// BurstCacheKey is a key for a burst cache resource. Needs to implement the one
+// method --- String() --- used for turning the key into an LRU and LockTab key.
+type BurstCacheKey interface {
+	String() string
+}
+
+// NewBurstCache makes a new burst cache with the given size and cacheLife.
+// name is for debugging purposes.
+func NewBurstCache(g *GlobalContext, sz int, dur time.Duration, name string) *BurstCache {
+	lru, err := lru.New(sz)
+	if err != nil {
+		g.Log.Fatalf("Bad LRU Constructor: %s", err.Error())
```

Comment:
panic. `lru` is nil at this point so it's going to panic later anyway.

Reply:
doesn't fatalf panic?

Reply:
oh yeah. log.Fatalf says it logs then os.Exit, and log.Panicf logs then panic. Either's fine